File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/factory/transport/Plane.java
// Plane.java

package com.bigboxcode.designpattern.factory.transport;

public class Plane implements Transport {
    @Override
    public void start() {
        System.out.println("Plane started");
    }

    @Override
    public void stop() {
        System.out.println("Plane Stopped");
    }

    @Override
    public void repair() {
        System.out.println("Plane Repair");
    }
}

File: design-patterns/ts/factory/transport-factory-method/plane.ts
// plane.ts

import Transport from "../transport/transport";

class Plane implements Transport {
    start(): void {
        console.log("Plane started");
    }

    stop(): void {
        console.log("Plane Stopped");
    }

    repair(): void {
        console.log("Plane Repair");
    }
}

export default Plane;